ext/pgsql/pgsql.c

```diff
@@ -2495,19 +2481,17 @@ PHP_FUNCTION(pg_lo_open)
 	   faster to type. Unfortunately, doesn't behave the same way as fopen()...
 	   (Jouni)
 	*/
-
-	if (strchr(mode_string, 'r') == mode_string) {
+	if (zend_string_equals_literal(mode, "r")) {
 		pgsql_mode |= INV_READ;
-		if (strchr(mode_string, '+') == mode_string+1) {
-			pgsql_mode |= INV_WRITE;
-		}
-	}
-	if (strchr(mode_string, 'w') == mode_string) {
+	} else if (zend_string_equals_literal(mode, "w")) {
 		pgsql_mode |= INV_WRITE;
-		create = 1;
-		if (strchr(mode_string, '+') == mode_string+1) {
-			pgsql_mode |= INV_READ;
-		}
+		create = true;
```

Comment:
Aren't you still missing create=true for the `w+` case?

Reply:
Indeed, this does make this whole block still as messy as before but probably still clearer